### DISCUSSION TITLE
Blocks - templating and titling

### DISCUSSION THREAD

**Discussion by josephwright** [josephwright/ltx-talk#13] [General]
In `beamer`, the `block` environment takes a title and may be used for design purposes. However, it is in many ways the same in concept as the L3 `block` template, which is used in the current development code to re-implement quotes, lists, etc. This leads to a few related questions

- Should `xbeamer` call this environment `block` at all?
- If so, should such blocks be implemented using the L3 `block` template system? The would currently mean requiring the `latex-lab` code (`\DocumentMetadata{}` line)
- How does this impact on the requirement for a block title?

**Comment by josephwright** [comment]
Answering my own questions with my current feeling

>     * Should `xbeamer` call this environment `block` at all?

Yes, I think `block` is a good name for a generic container in a presentation: we might imagine doing 'design' here but also in any other document

>     * If so, should such blocks be implemented using the L3 `block` template system? The would currently mean requiring the `latex-lab` code (`\DocumentMetadata{}` line)

Again, I think 'yes' once we go beyond the current stub.

>     * How does this impact on the requirement for a block title?

My feeling is that this is similar to `\frametitle`, so I'm minded to drop the argument and have `\blocktitle`.

**Comment by samcarter** [comment]
I like the name `block`, user will know what it is from beamer.

---

It is certainly the same situation as with the frame title, so a consistent solution makes sense. 

Beamer is a bit weird when it comes to an empty block title. For the rounded theme, it will throw away an empty title and just show the block body. For the rectangular theme, it will show an empty block title. There have been many questions from users who wanted the behaviour of whatever inner theme they didn't use, so both approaches seem to have fans.

With `\blocktitle` one would have the chance to let the user control this. No `\blocktitle` in the block could completely hide the title, `\blocktitle{}` could show an empty title bar. So `\blocktitle` would not only be consistent, but also offer more functionality.

**Comment by u-fischer** [comment]
I would require `\DocumentMetadata` and directly try to use as much of the new code as possible.  It will simplify coding. I mean you do not want to have to handle two list definitions  when you add overlay support.

**Comment by josephwright** [comment]
@samcarter  Very good point: I've always found it awkward that I have to have a title, as I have used blocks in the past as arbitrary containers.

**Comment by josephwright** [comment]
Also been thinking about tagging. I've made the block title a subsection for the present, but it's likely better considered in a presentation as some kind of caption. (That seems to contrast with a poster, where people do use blocks as more-or-less sections in my experience.)

**Comment by samcarter** [comment]
Just as a data point for comparison: Rmarkdown uses the heading level below the frametitle for block titles, e.g.

```
---
output: beamer_presentation
---

# 1

test

## test
```

will give

```
\begin{frame}{1}
\phantomsection\label{section}
test

\begin{block}{test}
\phantomsection\label{test}
\end{block}
\end{frame}
```

**Comment by Skillmon** [comment]
But in `beamer` sections and subsections are parallel to frametitles etc.

Also I know presentations in which the frame title was the section, and the subtitle the subsection plus the actual frame title (some of my professors used such a style to have their presentations in sync with scripts etc.).

**Comment by josephwright** [comment]
@Skillmon  Well yes it depends very much how one uses this: Till provided sections as well as frame/block names so one can have structure using 'standard' sectioning as well (and there's `lecture`, which is a bit like `part` ...). The thing here is it's not obvious whether one can know how 'section like' titles are, at least in a generic sense.

**Comment by josephwright** [comment]
I think for the present we will stick with the model that blocks are like frame and have an explicit `\blocktitle` command - different from `beamer` but overall more consistent.
